File: DotNetMessaging/backend/DotNetMessaging.API/DTOs/ChatDTOs.cs
namespace DotNetMessaging.API.DTOs;

public class ChatDto
{
    public string Id { get; set; } = string.Empty;
    public UserDto OtherUser { get; set; } = null!;
    public MessageDto? LastMessage { get; set; }
    public long UnreadCount { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class GroupDto
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ProfilePictureUrl { get; set; }
    public List<GroupMemberDto> Members { get; set; } = new();
    public MessageDto? LastMessage { get; set; }
    public long UnreadCount { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class GroupMemberDto
{
    public string Id { get; set; } = string.Empty;
    public UserDto User { get; set; } = null!;
    public string Role { get; set; } = string.Empty;
    public DateTime JoinedAt { get; set; }
}

public class CreateGroupRequest
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public List<string> MemberIds { get; set; } = new();
}

File: DotNetMessaging/SpringBootMessaging/src/main/java/com/messaging/api/dtos/ChatDTOs.java
package com.messaging.api.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

public class ChatDTOs {
    
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class ChatDto {
        private String id;
        private AuthDTOs.UserDto otherUser;
        private MessageDto lastMessage;
        private long unreadCount;
        private LocalDateTime createdAt;
    }
    
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class GroupDto {
        private String id;
        private String name;
        private String description;
        private String profilePictureUrl;
        private List<GroupMemberDto> members;
        private MessageDto lastMessage;
        private long unreadCount;
        private LocalDateTime createdAt;
    }
    
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class GroupMemberDto {
        private String id;
        private AuthDTOs.UserDto user;
        private String role;
        private LocalDateTime joinedAt;
    }
    
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class CreateGroupRequest {
        private String name;
        private String description;
        private List<String> memberIds;
    }
}

